rust/tool/src/esp.rs

```diff
@@ -131,10 +152,11 @@ mod tests {
         let path =
             Path::new("/nix/store/xqplddjjjy1lhzyzbcv4dza11ccpcfds-initrd-linux-6.1.1/initrd");
 
-        let generated_filename = nixos_path(path, "initrd")?;
+        let generated_filename = path_with_hash([("initrd", path)], "initrd")?;
 
-        let expected_filename =
-            PathBuf::from("xqplddjjjy1lhzyzbcv4dza11ccpcfds-initrd-linux-6.1.1-initrd.efi");
+        let expected_filename = PathBuf::from(
+            "2eb997ca2f81d4216b6d594b7dc00c9fd9d13c7e7a4ea6ce17a92c19e0b81944-initrd.efi",
```

Comment:
This make us lose the capability to read the kernel version on the filename, I personally use this feature but I don't know if we need it.